jax/_src/lax/control_flow/for_loop.py

```diff
@@ -0,0 +1,262 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module for the `for_loop` primitive."""
+from functools import partial
+
+from typing import Any, List, Tuple
+
+from jax import core
+from jax.interpreters import ad
+from jax._src import ad_util
+from jax._src import pretty_printer as pp
+
+
+## Helpful type aliases
+Ref = Array = Any
+
+## State effect
+
+State = object()
+
+## get/swap/addupdate implementations
+
+# `get` reads a value reading a value from a `Ref` type, a.k.a.:
+# a <- x[] -- a <=> get_p.bind(x, idx=())
```

Comment:
I'm not sure about these comments; the the thing to the left of `<=>` isn't jaxpr syntax, and it's not Python syntax, so what's it meant to be? And then what's `<=>` meant to mean exactly? (And is the `-- a` a typo?)

My guess is it's meant to be jaxpr syntax, but in that case I'd revise it to be more precise, like

```
# Staging out `a = get_p.bind(x, idx=())` where the aval of `x` is
# `ShapedArrayRef((3,), np.dtype('float32'))` leads to a jaxpr eqn printed like
#   b:f32[] <- x[]
```

WDYT?

Reply:
If the comment is about jaxpr pretty-printing, maybe it should be moved down to the pretty-printing section?

If it's instead about the semantics of the get operation and maybe its idx parameter, maybe it should be defined without referring to pseudo-syntax (i.e. the `a <- x[]` stuff) and instead using words?

Reply:
Yep, good idea.